modules/azure/src/test/java/org/testcontainers/azure/AzureServiceBusEmulatorContainerTest.java

```diff
@@ -0,0 +1,102 @@
+package org.testcontainers.azure;
+
+import com.azure.messaging.servicebus.ServiceBusClientBuilder;
+import com.azure.messaging.servicebus.ServiceBusErrorContext;
+import com.azure.messaging.servicebus.ServiceBusMessage;
+import com.azure.messaging.servicebus.ServiceBusProcessorClient;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessage;
+import com.azure.messaging.servicebus.ServiceBusReceivedMessageContext;
+import com.azure.messaging.servicebus.ServiceBusSenderClient;
+import com.azure.messaging.servicebus.ServiceBusTransactionContext;
+import com.github.dockerjava.api.model.Capability;
+import org.assertj.core.api.Assertions;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testcontainers.containers.MSSQLServerContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.utility.DockerImageName;
+import org.testcontainers.utility.MountableFile;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AzureServiceBusEmulatorContainerTest {
+
+    @Rule
+    // network {
+    public Network network = Network.newNetwork();
+
+    // }
+
+    @Rule
+    // sqlContainer {
+    public MSSQLServerContainer<?> mssqlServerContainer = new MSSQLServerContainer<>(
+        "mcr.microsoft.com/mssql/server:2022-CU14-ubuntu-22.04"
+    )
+        .acceptLicense()
+        .withPassword("yourStrong(!)Password")
+        .withCreateContainerCmdModifier(cmd -> {
+            cmd.getHostConfig().withCapAdd(Capability.SYS_PTRACE);
+        })
+        .withNetwork(network);
+
+    // }
+
+    @Rule
+    // emulatorContainer {
+    public AzureServiceBusEmulatorContainer emulator = new AzureServiceBusEmulatorContainer(
+        DockerImageName.parse("mcr.microsoft.com/azure-messaging/servicebus-emulator:1.0.1"),
+        mssqlServerContainer
+    )
+        .acceptLicense()
+        .withConfig(MountableFile.forClasspathResource("/service-bus-config.json"))
+        .withNetwork(network);
+
```

Comment:
Let's keep the constructor only accepting the image as a String or DockerImageName. Rather we can declare dependsOn via de API.

```suggestion
    public AzureServiceBusEmulatorContainer emulator = new AzureServiceBusEmulatorContainer(
        DockerImageName.parse("mcr.microsoft.com/azure-messaging/servicebus-emulator:1.0.1")
    )
        .acceptLicense()
        .withConfig(MountableFile.forClasspathResource("/service-bus-config.json"))
        .withNetwork(network)
        .dependsOn(mssqlServerContainer);
```

Reply:
I have created a new `withMsSqlServerContainer(MSSQLServerContainer)` method instead because I needed to get the password and the network alias for the env setup in `configure()`. For this reason the dependency on the module remained as-is too. Is this acceptable?

Reply:
totally fine 👍🏽